app/components/homepage-components/WorkExperienceCard.tsx

```diff
@@ -0,0 +1,58 @@
+"use client";
+
+import React, { useState } from "react";
+import Image from "next/image";
+import { useTheme } from "next-themes";
+
+interface Job {
+  title: string;
+  position: string;
+  duration: number;
+  durationUnit: string;
+  image: string;
+  imageDark: string;
+  imagePrimaryLight: string;
+  imagePrimaryDark: string;
+}
+
+interface WorkExperienceCardProps {
+  job: Job;
+}
+
+const WorkExperienceCard: React.FC<WorkExperienceCardProps> = ({ job }) => {
+  const [isMouseOver, setIsMouseOver] = useState(false);
+  const { theme } = useTheme();
+  return (
+    <div
+      className={`flex flex-col rounded-xl border border-white800 bg-white900 p-9 dark:border-0 dark:bg-black200 xl:flex-row ${
+        isMouseOver && "shadow-lg dark:bg-black300"
+      }`}
+      onMouseOver={() => setIsMouseOver(true)}
+      onMouseLeave={() => setIsMouseOver(false)}
+    >
+      <Image
+        src={
+          theme === "light"
+            ? isMouseOver
+              ? job.imagePrimaryLight
+              : job.image
+            : isMouseOver
+            ? job.imagePrimaryDark
+            : job.imageDark
+        }
```

Comment:
This nested ternary is a little unclear.

Consider setting a variable to determine this state if this the route you'd like to go:

```js
// above the return:
let imageSrc;

if (theme === "light") {
  imageSrc = isMouseOver ? job.imagePrimaryLight : job.image;
} else {
  imageSrc = isMouseOver ? job.imagePrimaryDark : job.imageDark;
}

// then in your Image:
<Image src={imageSrc} />
```

You could also consider simplifying this by using one colored image and relying on CSS and 'hover' instead of states and multiple images. Here's an example:

https://play.tailwindcss.com/zq8Eg3o2mS

Reply:
I habe fixed this now and it is way better thank you for you suggestion